rpcserver.go

```diff
@@ -6570,3 +6600,31 @@ func encodeVirtualPackets(packets []*tappsbt.VPacket) ([][]byte, error) {
 
 	return rawPackets, nil
 }
+
+// EncodePrecisionInJSON takes a precision value and an encoded json object and
+// attempts to append a specific key-value pair in the json object, then encodes
+// it again and returns it.
+func EncodePrecisionInJSON(precision uint64, jBytes []byte) ([]byte, error) {
+	var jMeta map[string]interface{}
+
+	err := json.Unmarshal(jBytes, &jMeta)
```

Comment:
Will this fail for empty `jBytes`? Thinking of the case where the user sets nothing, and then specifies the precision field.

Reply:
nice catch, probably yeah, will test and adjust it accordingly
(we **should** be able to accept empty json)